docs/source/dynamo/custom-backends.rst

```diff
@@ -1,8 +1,95 @@
 Custom Backends
 ===============
 
+Overview
+--------
+
+``torch.compile`` provides a straightforward method to enable users
+to define custom backends.
+
+A backend function has the contract
+``(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor]) -> Callable``.
+
+Backend functions can be called by TorchDynamo, the graph tracing component of ``torch.compile``,
+after tracing an FX graph and are
+expected to return a compiled function that is equivalent to the traced FX graph.
+The returned callable should have the same contract as the ``forward`` function of an ``nn.Module``:
+``(*args: Any) -> Any``.
```

Comment:
This isn't exaclty right.  The contract will should match `gm: torch.fx.GraphModule` and will go from all tensor inputs to a list of tensors as output.